src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -796,47 +800,39 @@ private String getContainerCommand() {
 
     public void restartServer() throws PluginExecutionException {
+        info("Restarting server...");
+        setDevStop(true);
         if (container) {
-            // TODO for now stopServer and startServer do nothing if container==true, so really only the app is being redeployed.
-            // But eventually we should restart the container as well.
-            info("Redeploying application...");
+            stopContainer(); // this command is synchronous
         } else {
-            info("Restarting server...");
-        }
-        setDevStop(true);
-        stopServer();
-        if (serverThread != null) {
-            final long threadShutdownTimeoutSeconds = 30;
-            try {
-                serverThread.join(threadShutdownTimeoutSeconds * 1000);
-                if (serverThread.isAlive()) {
-                    throw new PluginExecutionException("Could not stop the server after " + threadShutdownTimeoutSeconds
-                            + " seconds.  Ensure that the server has been stopped, then start dev mode again.");
-                }
-            } catch (InterruptedException e) {
-                if (serverThread.isAlive()) {
-                    throw new PluginExecutionException(
-                            "Could not stop the server.  Ensure that the server has been stopped, then start dev mode again.",
-                            e);
-                } else {
-                    // the thread was interrupted, but the server thread is already stopped
-                    debug(e);
+            stopServer();
+            if (serverThread != null) {
+                final long threadShutdownTimeoutSeconds = 30;
+                try {
+                    serverThread.join(threadShutdownTimeoutSeconds * 1000);
+                    if (serverThread.isAlive()) {
+                        throw new PluginExecutionException("Could not stop the server after " + threadShutdownTimeoutSeconds
+                                + " seconds.  Ensure that the server has been stopped, then start dev mode again.");
+                    }
+                } catch (InterruptedException e) {
+                    if (serverThread.isAlive()) {
+                        throw new PluginExecutionException(
+                                "Could not stop the server.  Ensure that the server has been stopped, then start dev mode again.",
+                                e);
+                    } else {
+                        // the thread was interrupted, but the server thread is already stopped
+                        debug(e);
+                    }
                 }
             }
         }
         libertyCreate();
-        if (!container) {
-            // local dev mode can't install feature on containerized runtime, unless we exec into the container
-            libertyInstallFeature();
-        }
+        // On containerized runtime local dev mode can't install feature but install in local server for consistency
```

Comment:
For container mode, I think we should still skip installing features for now.  There's no need to install locally for consistency, since if dev mode is started locally afterwards then the local features will be installed at that time.

Reply:
Agreed.